src/Kiota.Builder/Writers/CSharp/CodeIndexerWriter.cs

```diff
@@ -13,6 +13,9 @@ public override void WriteCodeElement(CodeIndexer codeElement, LanguageWriter wr
         if (codeElement.Parent is not CodeClass parentClass) throw new InvalidOperationException("The parent of a property should be a class");
         var returnType = conventions.GetTypeString(codeElement.ReturnType, codeElement);
         conventions.WriteShortDescription(codeElement.Documentation.Description, writer);
+        var deprecationMessage = conventions.GetDeprecationInformation(codeElement);
+        if (!string.IsNullOrEmpty(deprecationMessage))
+            writer.WriteLine(deprecationMessage);
```

Comment:
I think this block of code has been repeated enough times in the C# code writers to warrant a function in the conventions class that is similar to the Go conventions
```cs
internal void WriteDeprecation(IDeprecableElement element, LanguageWriter writer)
```

Reply:
thanks for pointing this out! I started with CSharp and forgot to revisit the structure!